src/re_frame/interop.clj

```diff
@@ -0,0 +1,43 @@
+(ns re-frame.interop)
+
+(defn next-tick [f]
+  (.start (Thread. f)))
+
+(def empty-queue clojure.lang.PersistentQueue/EMPTY)
+
+(def after-render next-tick)
+
+(def debug-enabled? true)
+
+(defn ratom [x]
+  (let [result (atom x)]
+    (alter-meta! result assoc ::ratom? true)
+    result))
+
+(defn ratom? [maybe-ratom]
+  (and (satisfies? clojure.lang.IAtom maybe-ratom)
+       (::ratom? (meta maybe-ratom))))
+
+(defn make-reaction
+  "On JVM Clojure, return an atom-like thing which invokes the given function on
+  every deref. That is, `make-reaction` here provides precisely none of the
+  benefits of `reagent.ratom/make-reaction` (which only invokes its function if
+  the reactions that the function derefs have changed value). But so long as `f`
+  only depends on other reactions (which also behave themselves), the only
+  difference is one of efficiency. That is, your tests should see no difference
+  other than that they do redundant work."
+  [f]
+  (reify clojure.lang.IDeref
```

Comment:
Personally, I think it's better not to try do that at all.

Fundamentally, unless we actually re-implement the whole Reagent and React component tree rendering lifecycle (invoking Reagent render functions only when Reagent would, etc), I think it's going to be almost impossible to ensure we `deref` ratoms in tests only in the same way as Reagent would in the real world.  If we implement some form of `IReactiveAtom` and `IDisposable` regardless, then I think that's more likely to mislead people into thinking that we *are* accurately mimicking real-world behaviour, and writing bogus tests.

I think the naive implementation is fine, though, because as described in the docstring here, in most cases that won't make a difference.  For cases where it *would* make a difference, I think we just need to say "you can't test that type of behaviour with this test code".

Having said that, the sort of code I'm thinking of testing with this is all code for which this doesn't make a difference.  If you have some examples of code which would be not be accurately testable with the above, but which would become accurately testable with some additional cleverness in this `ns`, I'd be interested to work through what might be achievable here.

Regardless, at the very least I think your comment is enough to suggest that an more complete explanation for the approach this namespace takes is probably warranted.